src/window.rs

```diff
@@ -599,6 +599,15 @@ impl AppWindow {
     pub fn set_color(&self, color: Color) {
         if self.color() != Some(color) {
             let history_item = HistoryObject::new(color);
+            for (index, item) in self.history().snapshot().iter().enumerate() {
+                if Cast::downcast_ref::<HistoryObject>(item)
+                    .expect("Couldn't read history object")
+                    .color()
+                    == color.into()
+                {
+                    self.history().remove(index.try_into().unwrap());
+                }
+            }
```

Comment:
This should be replaced with [`find_with_equal_func`](https://docs.gtk.org/gio/method.ListStore.find_with_equal_func.html)